packages/@lwc/engine-core/src/framework/api.ts

```diff
@@ -588,7 +591,14 @@ function dc(
         );
     }
 
-    return null;
+    const sel = getComponentRegisteredName(Ctor);
```

Comment:
For my understanding: The sel needs to be looked up from the `Ctor` here, at runtime, because the sel depends on the `Ctor`'s metadata(name & namespace) and the template-compiler does not have this info ahead of time. Add a comment to that effect 🙂

Reply:
Good call, added some [comments](https://github.com/salesforce/lwc/pull/3338/commits/ad169dab4fac866e8ac918d773b3d9cc75e85594#diff-a284e53a17a2b48cc32e6bf3b121b3acb849a57df886d1cb92a2a46df5aebb7bR592-R595).  Let me know if it's clear enough.